src/swap/defi/lifi.ts

```diff
@@ -319,6 +325,51 @@ export function makeLifiPlugin(opts: EdgeCorePluginOptions): EdgeSwapPlugin {
     let preTx: EdgeTransaction | undefined
     let spendInfo: EdgeSpendInfo
     switch (fromWallet.currencyInfo.pluginId) {
+      case 'solana': {
```

Comment:
I see. We don't do token approvals on Solana, so this has to be pretty different.